libminifi/test/persistence-tests/PersistenceTests.cpp

```diff
@@ -0,0 +1,218 @@
+
+    // setup MERGE processor
+    {
+      merge = std::make_shared<MergeContent>("MergeContent", mergeProcUUID());
+      merge->initialize();
+      merge->setAutoTerminatedRelationships({{"original", "d"}});
+
+      merge->setProperty(MergeContent::MergeFormat, MERGE_FORMAT_CONCAT_VALUE);
+      merge->setProperty(MergeContent::MergeStrategy, MERGE_STRATEGY_BIN_PACK);
+      merge->setProperty(MergeContent::DelimiterStrategy, DELIMITER_STRATEGY_TEXT);
+      merge->setProperty(MergeContent::MinEntries, "3");
+      merge->setProperty(MergeContent::Header, "_Header_");
+      merge->setProperty(MergeContent::Footer, "_Footer_");
+      merge->setProperty(MergeContent::Demarcator, "_Demarcator_");
+      merge->setProperty(MergeContent::MaxBinAge, "1 h");
+
+      std::shared_ptr<core::ProcessorNode> node = std::make_shared<core::ProcessorNode>(merge);
+      mergeContext = std::make_shared<core::ProcessContext>(node, controller_services_provider, prov_repo, ff_repository, content_repo);
+    }
+
+    // setup INPUT processor
+    {
+      inputProcessor = std::make_shared<core::Processor>("source", inputProcUUID());
+      std::shared_ptr<core::ProcessorNode> node = std::make_shared<core::ProcessorNode>(inputProcessor);
+      inputContext = std::make_shared<core::ProcessContext>(node, controller_services_provider, prov_repo,
+                                                            ff_repository, content_repo);
+    }
+
+    // setup Input Connection
+    {
+      input = std::make_shared<Connection>(ff_repository, content_repo, "Input", inputConnUUID());
+      input->setRelationship({"input", "d"});
+      input->setDestinationUUID(mergeProcUUID());
+      input->setSourceUUID(inputProcUUID());
+      inputProcessor->addConnection(input);
+    }
+
+    // setup Output Connection
+    {
+      output = std::make_shared<Connection>(ff_repository, content_repo, "Output", outputConnUUID());
+      output->setRelationship(MergeContent::Merge);
+      output->setSourceUUID(mergeProcUUID());
+    }
+
+    // setup ProcessGroup
+    {
+      root = std::make_shared<core::ProcessGroup>(core::ProcessGroupType::ROOT_PROCESS_GROUP, "root");
+      root->addProcessor(merge);
+      root->addConnection(input);
+      root->addConnection(output);
+    }
+
+    // prepare Merge Processor for execution
+    merge->setScheduledState(core::ScheduledState::RUNNING);
+    merge->onSchedule(mergeContext.get(), new core::ProcessSessionFactory(mergeContext));
+  }
+  void write(const std::string& data) {
+    minifi::io::DataStream stream(reinterpret_cast<const uint8_t*>(data.c_str()), data.length());
+    core::ProcessSession sessionGenFlowFile(inputContext);
+    std::shared_ptr<core::FlowFile> flow = std::static_pointer_cast<core::FlowFile>(sessionGenFlowFile.create());
+    sessionGenFlowFile.importFrom(stream, flow);
+    sessionGenFlowFile.transfer(flow, {"input", "d"});
+    sessionGenFlowFile.commit();
```

Comment:
I think assertions on the `ResourceClaim` refcount would be useful documentation here.

Reply:
done